pkg/core/storage/store.go

```diff
@@ -96,6 +96,11 @@ type (
 		// Key and value slices should not be modified.
 		// Seek can guarantee that key-value items are sorted by key in ascending way.
 		Seek(rng SeekRange, f func(k, v []byte) bool)
+		// SeekGC is similar to Seek, but the function should return true if current
+		// KV pair should be kept and false if it's to be deleted; there is no way to
+		// do an early exit here. SeekGC only works with the current Store, it won't
+		// go down to layers below and it takes a full write lock, so use it carefully.
+		SeekGC(rng SeekRange, keep func(k, v []byte) bool) error
```

Comment:
And maybe it's better to rename standard `Seek(rng SeekRange, f func(k, v []byte) bool)`  function argument? I.e. to have `Seek(rng SeekRange, cont func(k, v []byte) bool)`.

Reply:
Well, we've had `f` for a long time, I think it's OK.